Chore/rename task fields

luisaadanttas opened the pull request:
This pull request addresses issue #938

yoavkatz commented:
Thanks for your contribution Maria! The changes look right.  However,   the proposed change is not backward compatible (require changes to user code). Since we don't want to make abrupt changes, we should keep the original fields in Task , but add them as deprecated.  Then we should check in the Task. prepare() that either the old fields or new fields are set (but not both).  It's good that you changed all the existing documentation and examples, so people will follow the new approach from now on. Finally, we should add one test that uses the old fields and see it continues works.

Do you think you can proceed with these changes.

luisaadanttas commented:
Yes, that makes a lot of sense. Thank you!

I can implement those changes :)

elronbandel commented:
@luisaadanttas thanks again! notice you mark the old fields with DeprectedField.
See this example for reference:
https://github.com/IBM/unitxt/blob/279412f6650eb69a4a9b7769100b65a31453ed7f/src/unitxt/operators.py#L257

yoavkatz commented on src/unitxt/task.py:
```diff
@@ -38,20 +39,49 @@ class Task(InstanceOperator):
+        )
+
     def verify(self):
         for io_type in ["input_fields", "reference_fields"]:
```
We should check that input_fields and reference_fields are not None (because they are optional fields).
Best to check each on separately:

 if (self.input_fields is   None):
            raise ValueError(
                "Missing attribute in task attributes 'input_fields' not set."
            )
 if (self.reference_fields is   None):
            raise ValueError(
                "Missing attribute: 'reference_fields' not set."
            )

yoavkatz commented on src/unitxt/task.py:
```diff
@@ -38,20 +39,49 @@ class Task(InstanceOperator):
 
+    def prepare(self):
+        super().prepare()
+        if (self.input_fields is not None and self.reference_fields is not None) and (
```
I suggest separating into two checks, because people can mix only one pair of fields.

if (self.input_fields is not None and self.inputs is not None):
  aise ValueError(
                "Conflicting attributes: 'input_fields'  cannot be set simultaneously with 'inputs'. Use only 'input_fields'"
            )

yoavkatz reviewed (changes requested):
Thanks for making the changes. I added two minor comments regarding error checking - and then I think we are good to merge.

luisaadanttas commented:
> Thanks for making the changes. I added two minor comments regarding error checking - and then I think we are good to merge.

Thanks for your review and suggestions! I've incorporated your feedback on the field checks and updated the tests accordingly :)

yoavkatz commented:
Thanks for making the changes.  I think you need to run:
utils/prepare_all_artifacts.py to update a few json files that were not converted.

yoavkatz commented:
Changes are merged.

Thank you Maria for your contribution!
